pageserver/src/page_service.rs

```diff
@@ -1128,6 +1129,7 @@ impl PageServerHandler {
                                 // Closing the connection by returning ``::Reconnect` has the side effect of rate-limiting above message, via
                                 // client's reconnect backoff, as well as hopefully prompting the client to load its updated configuration
                                 // and talk to a different pageserver.
+                                MISROUTED_PAGESTREAM_REQUESTS.inc();
```

Comment:
This metric is being bumped multiple times for the same request. I think it makes sense to remove the increment from resolve and keep just this one (this is when we actually tell the compute to reconnect).